endpoints/openrtb2/auction_test.go

```diff
@@ -2912,6 +2912,108 @@ func TestValidateNativeAssetData(t *testing.T) {
+		analyticsConf.NewPBSAnalytics(&config.Analytics{}),
+		map[string]string{},
+		false,
+		[]byte{},
+		openrtb_ext.BuildBidderMap(),
+		nil,
+		nil,
+		hardcodedResponseIPValidator{response: true},
+	}
+
+	if specFiles, err := ioutil.ReadDir("../../firstpartydata/tests/holdauction"); err == nil {
+		for _, specFile := range specFiles {
+			fileName := "../../firstpartydata/tests/holdauction/" + specFile.Name()
+			fpdFile, err := loadFpdFile(fileName)
+
+			if err != nil {
+				t.Errorf("Unable to load file: %s", fileName)
+			}
+
+			var resRequest openrtb2.BidRequest
+			err = json.Unmarshal(fpdFile.ResultRequestData, &resRequest)
+			if err != nil {
+				t.Errorf("Unable to unmarshal input request: %s", fileName)
+			}
+
+			reqBody := string(fpdFile.InputRequestData)
+			deps.cfg = &config.Configuration{MaxRequestSize: int64(len(reqBody))}
+
+			req := httptest.NewRequest("POST", "/openrtb2/auction", strings.NewReader(reqBody))
+			recorder := httptest.NewRecorder()
+
+			deps.Auction(recorder, req, nil)
+
+			if recorder.Code != http.StatusOK {
+
+				resp, err := ioutil.ReadAll(recorder.Body)
+				assert.NoError(t, err, "Error shuld not be nil")
+
+				for i := range fpdFile.ValidationErrors {
+					assert.Contains(t, string(resp), fpdFile.ValidationErrors[i], "Incorrect first party data warning message")
+				}
+				continue
+			}
+
+			actualRequest := deps.ex.(*warningsCheckExchange).auctionRequest
+
+			assert.Equal(t, len(actualRequest.FirstPartyData), len(fpdFile.ResultBiddersFPD))
+
+			for k, expectedValue := range fpdFile.ResultBiddersFPD {
+				actualValue := actualRequest.FirstPartyData[k]
+
+				if expectedValue.Site != nil {
+					assert.Equal(t, expectedValue.Site, actualValue.Site, "Incorrect first party data")
+					assert.Equal(t, resRequest.Site, actualRequest.BidRequest.Site, "Incorrect site in request")
+				}
+				if expectedValue.App != nil {
+					assert.Equal(t, expectedValue.App, actualValue.App, "Incorrect first party data")
+					assert.Equal(t, resRequest.App, actualRequest.BidRequest.App, "Incorrect app in request")
+				}
+				if expectedValue.User != nil {
+					assert.Equal(t, expectedValue.User, actualValue.User, "Incorrect first party data")
+					assert.Equal(t, resRequest.User, actualRequest.BidRequest.User, "Incorrect user in request")
+				}
+			}
+
+		}
+	}
+}
+
+func loadFpdFile(filename string) (fpdFile, error) {
+	var fileData fpdFile
+	fileContents, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return fileData, err
+	}
+	err = json.Unmarshal(fileContents, &fileData)
+	if err != nil {
+		return fileData, err
+	}
+
+	return fileData, nil
+}
+
+type fpdFile struct {
+	InputRequestData  json.RawMessage                               `json:"inputRequestData,omitempty"`
+	ResultRequestData json.RawMessage                               `json:"resultRequestData,omitempty"`
+	InputBiddersFPD   map[openrtb_ext.BidderName]*openrtb_ext.ORTB2 `json:"inputBiddersFPD,omitempty"`
```

Comment:
I don't see `InputBiddersFPD` being used anywhere. Can we please remove this?

Reply:
Yes, removed